### PR TITLE
Add "fromMe" option to fetchMessages

### PR DESCRIPTION
**Pull request by roigreenberg**
I added an option to get only messages from the bot, or not from the bot.
This is needed in some edge case where there is a message from the bot after calling "countUnread". otherwise, fetchMessages will return the bot message instead a unread message.

### CODE REVIEW

**Review by purpshell — CHANGES_REQUESTED** [review]
Hey! please rebase your PR and remove anything not related to the "fromMe" option, thanks. will approve after that

**Comment by roigreenberg** [comment]
@PurpShell I removed any unrelated changes, I hope that is enough.

File: src/structures/Chat.js [hunk 1]
```diff
@@ -170,13 +170,24 @@ class Chat extends Base {
...
      */
     async fetchMessages(searchOptions) {
         let messages = await this.client.pupPage.evaluate(async (chatId, searchOptions) => {
-            const msgFilter = m => !m.isNotification; // dont include notification messages
+            const msgFilter = (m) => {
+                if (m.isNotification) {
+                    return false; // dont include notification messages
+                }
+                if (searchOptions) {
+                    if (searchOptions.fromMe !== undefined && m.id.fromMe !== searchOptions.fromMe) {
```
Comment (PurpShell) [inline src/structures/Chat.js:185]:
```suggestion
                    if (searchOptions.fromMe && m.id.fromMe !== searchOptions.fromMe) {
```

**Review by purpshell — APPROVED** [review]

**Comment by PurpShell** [comment]
lgtm

File: src/structures/Chat.js [hunk 2]
```diff
@@ -170,13 +170,24 @@ class Chat extends Base {
...
      */
     async fetchMessages(searchOptions) {
         let messages = await this.client.pupPage.evaluate(async (chatId, searchOptions) => {
-            const msgFilter = m => !m.isNotification; // dont include notification messages
+            const msgFilter = (m) => {
+                if (m.isNotification) {
+                    return false; // dont include notification messages
+                }
+                if (searchOptions) {
+                    if (searchOptions.fromMe && m.id.fromMe !== searchOptions.fromMe) {
```
Comment (tuyuribr) [inline src/structures/Chat.js:185]: LGTM but put searchOptions inside the inner if. (searchOptions && searchOptions.fromMe && .... )

**Comment by tuyuribr** [comment]
Also don't forget to document it in **index.d.ts**

**Review by tuyuribr — APPROVED** [review]

**Pull request merged**